bot/utils/blurple_formatter.py

```diff
@@ -0,0 +1,717 @@
+        else:
+            yield f"{n * ' '}{f'{space()}'.join(curr_chunk)}"
+            yield from (f"{n * ' '}{line}" for line in stmt_lines)
+            curr_len = 0
+            curr_chunk = []
+    yield f"{n * ' '}{f'{space()}'.join(curr_chunk)}"
+
+
+def unparse(node: ast.AST, nl_able: bool = False) -> str:
+    """Convert the ast node to formatted(!) Python source code."""
+    if isinstance(node, ast.AnnAssign):
+        target, annotation, value = node.target, node.annotation, node.value
+
+        return (
+            f"{unparse(target)}{space()}:{space()}{unparse(annotation)}"
+            + (f"{space()}={space()}{unparse(value)}" if value else "")
+            + f"{space()};"
+        )
+    if isinstance(node, ast.Assert):
+        test, msg = node.test, node.msg
+
+        return (
+            f"assert{space(1)}{test}"
+            + (f",{space()}{unparse(msg)}" if msg else "")
+            + f"{space()};"
+        )
+    if isinstance(node, ast.Assign):
+        targets, value = node.targets, node.value
+
+        return (
+            f"{f'{space()}={space()}'.join(map(unparse, targets))}{space()}={space()}{unparse(value)}"
+            + f"{space()};"
+        )
+    if isinstance(node, ast.AsyncFor):
+        target, iter_, body, orelse = node.target, node.iter, node.body, node.orelse
+
+        s = [
+            f"async{space()}for{space()}{unparse(target)}{space()}in{space()}{unparse(iter_)}{space()}:",
+            *indent(body, randrange(2, 5)),
+        ]
+        if orelse:
+            s.append(f"else{space()}:")
+            s.extend(indent(orelse, randrange(2, 5)))
+        return "\n".join(s)
+    if isinstance(node, ast.AsyncFunctionDef):
+        name, args, body, decorator_list, returns = (
+            node.name,
+            node.args,
+            node.body,
+            node.decorator_list,
+            node.returns,
+        )
+
+        return "\n".join(
+            (
+                *(f"@{space()}{unparse(decorator)}" for decorator in decorator_list),
+                f"async{space(1)}def{space(1)}{name}{space()}({space()}{unparse(args, True)}{space()})"
+                + (f"{space()}->{space()}{unparse(returns)}" if returns else "")
+                + f"{space()}:",
+                *indent(body, randrange(2, 5)),
+            )
+        )
+    if isinstance(node, ast.AsyncWith):
+        items, body = node.items, node.body
+
+        return "\n".join(
+            [
+                f"async{space(1)}with {f',{space()}'.join(map(unparse, items))}{space()}:",
+                *indent(body, randrange(2, 5)),
+            ]
+        )
+    if isinstance(node, ast.Attribute):
+        owner, attr = node.value, node.attr
+
+        return f"{unparse(owner)}{space()}.{space()}{attr}"
+    if isinstance(node, ast.AugAssign):
+        target, op, value = node.target, node.op, node.value
+
+        return (
+            f"{unparse(target)}{space()}{op_strs[type(op)]}={space()}{unparse(value)};"
```

Comment:
We should be able to have space here right?
```suggestion
            f"{unparse(target)}{space()}{op_strs[type(op)]}{space()}={space()}{unparse(value)};"
```

Additionally, shouldn't we have spaces before `;`? That's what you have for some of the statements.